spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/store/JdbcChannelMessageStore.java

```diff
@@ -448,23 +462,25 @@ public MessageGroup getMessageGroup(Object groupId) {
 	@ManagedAttribute
 	public int getMessageGroupCount() {
 		return this.jdbcTemplate.queryForObject(
-				getQuery("SELECT COUNT(DISTINCT GROUP_KEY) from %PREFIX%CHANNEL_MESSAGE where REGION = ?"),
+				getQuery(Query.COUNT_GROUPS,
+						() -> "SELECT COUNT(DISTINCT GROUP_KEY) from %PREFIX%CHANNEL_MESSAGE where REGION = ?"),
 				Integer.class, this.region);
 	}
 
 	/**
 	 * Replace patterns in the input to produce a valid SQL query. This implementation lazily initializes a
 	 * simple map-based cache, only replacing the table prefix on the first access to a named query. Further
 	 * accesses will be resolved from the cache.
-	 * @param sqlQuery The SQL query to be transformed.
+	 * @param queryName The {@link Query} to be transformed.
+	 * @param queryProvider a supplier to provide the query template.
 	 * @return A transformed query with replacements.
 	 */
-	protected String getQuery(String sqlQuery) {
-		String query = this.queryCache.get(sqlQuery);
+	protected String getQuery(Query queryName, Supplier<String> queryProvider) {
+		String query = this.queryCache.get(queryName);
```

Comment:
Why doesn't `computeIfAbsent()` work for us here?

Reply:
It does work; will fix; I simply changed the existing code.